ProcessLib/HydroMechanics/HydroMechanicsProcess.h

```diff
@@ -0,0 +1,162 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESS_LIB_HYDROMECHANICSPROCESS_H_
+#define PROCESS_LIB_HYDROMECHANICSPROCESS_H_
+
+#include <cassert>
+
+#include "MeshLib/Elements/Utils.h"
+#include "ProcessLib/Process.h"
+#include "ProcessLib/HydroMechanics/CreateLocalAssemblers.h"
+
+#include "HydroMechanicsFEM.h"
+#include "HydroMechanicsProcessData.h"
+
+namespace ProcessLib
+{
+namespace HydroMechanics
+{
+template <int DisplacementDim>
+class HydroMechanicsProcess final : public Process
+{
+    using Base = Process;
+
+public:
+    HydroMechanicsProcess(
+        MeshLib::Mesh& mesh,
+        std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&&
+            jacobian_assembler,
+        std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+        unsigned const integration_order,
+        std::vector<std::reference_wrapper<ProcessVariable>>&&
+            process_variables,
+        HydroMechanicsProcessData<DisplacementDim>&& process_data,
+        SecondaryVariableCollection&& secondary_variables,
+        NumLib::NamedFunctionCaller&& named_function_caller)
+        : Process(mesh, std::move(jacobian_assembler), parameters,
+                  integration_order, std::move(process_variables),
+                  std::move(secondary_variables),
+                  std::move(named_function_caller)),
+          _process_data(std::move(process_data))
+    {
+    }
+
+    //! \name ODESystem interface
+    //! @{
+
+    bool isLinear() const override { return false; }
+    //! @}
+
+private:
+    using LocalAssemblerInterface = HydroMechanicsLocalAssemblerInterface;
+
+void constructDofTable() override
+{
+    // Create single component dof in every of the mesh's nodes.
+    _mesh_subset_all_nodes.reset(
+        new MeshLib::MeshSubset(_mesh, &_mesh.getNodes()));
+    // Create single component dof in the mesh's base nodes.
+    _base_nodes = MeshLib::getBaseNodes(_mesh.getElements());
+    _mesh_subset_base_nodes.reset(
+        new MeshLib::MeshSubset(_mesh, &_base_nodes));
+
+    // Collect the mesh subsets in a vector.
+
+    // For pressure, which is the first
+    std::vector<std::unique_ptr<MeshLib::MeshSubsets>> all_mesh_subsets;
+    all_mesh_subsets.push_back(std::unique_ptr<MeshLib::MeshSubsets>{
+        new MeshLib::MeshSubsets{_mesh_subset_base_nodes.get()}});
+
+    // For displacement.
+    std::generate_n(
+        std::back_inserter(all_mesh_subsets),
+        _process_variables[1].get().getNumberOfComponents(),
+        [&]() {
+            return std::unique_ptr<MeshLib::MeshSubsets>{
+                new MeshLib::MeshSubsets{_mesh_subset_all_nodes.get()}};
+        });
+
+    _local_to_global_index_map.reset(new NumLib::LocalToGlobalIndexMap(
```

Comment:
you need to pass `std::vector<unsigned> const vec_n_components{1, DisplacementDim};` to the LocalToGlobalIndexMap constructor